bundles/org.openhab.binding.chatgpt/src/main/java/org/openhab/binding/chatgpt/internal/hli/ChatGPTHLIService.java

```diff
@@ -0,0 +1,429 @@
+                        ChatMessage message = new ChatMessage();
+                        message.setRole(ChatMessage.Role.TOOL.value());
+                        message.setName(functionName);
+                        message.setToolCallId(tool.getId());
+                        message.setContent(resultString);
+                        messages.add(message);
+                    } else {
+                        logger.debug("Function '{}' not found", functionName);
+                    }
+                }
+            }
+        });
+    }
+
+    private @Nullable String prepareRequestBody(String message) {
+        if (this.config == null) {
+            if (thingHandler instanceof ChatGPTHandler chatGPTHandler) {
+                this.config = chatGPTHandler.getConfigAs();
+            }
+        }
+
+        if (this.config == null) {
+            logger.debug("Could not get configuration");
+            return null;
+        }
+
+        LocalTime currentTime = LocalTime.now();
+        if (currentTime.isAfter(this.lastMessageTime.plusMinutes(config.keepContext))) {
+            this.messages.clear();
+        }
+        if (this.messages.isEmpty()) {
+            ChatMessage systemMessage = generateSystemMessage();
+            this.messages.add(systemMessage);
+        }
+
+        this.lastMessageTime = currentTime;
+        ChatMessage userMessage = new ChatMessage();
+        userMessage.setRole(ChatMessage.Role.USER.value());
+        userMessage.setContent(message);
+        this.messages.add(userMessage);
+        ChatRequestBody chatRequestBody = new ChatRequestBody();
+
+        if (this.config.chatGPTModel == null || this.config.chatGPTModel.isEmpty()) {
+            logger.debug("Model is not set");
+            return null;
+        }
+
+        chatRequestBody.setModel(this.config.chatGPTModel);
+        chatRequestBody.setTemperature(this.config.temperature);
+        chatRequestBody.setMaxTokens(this.config.maxTokens);
+        chatRequestBody.setTopP(this.config.topP);
+        chatRequestBody.setToolChoice(ToolChoice.AUTO.value());
+        chatRequestBody.setTools(this.tools);
+        chatRequestBody.setMessages(this.messages);
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.setSerializationInclusion(Include.NON_NULL);
+
+        try {
+            return objectMapper.writeValueAsString(chatRequestBody);
+        } catch (JsonProcessingException e) {
+            logger.debug("Failed to serialize ChatGPT request: {}", e.getMessage(), e);
+            return null;
+        }
+    }
+
+    private ChatMessage generateSystemMessage() {
+        ChatMessage systemMessage = new ChatMessage();
+        systemMessage.setRole(ChatMessage.Role.SYSTEM.value());
+        StringBuilder content = new StringBuilder();
+        content.append(this.config.systemMessage);
+
+        Collection<Item> openaiItems = itemRegistry.getItemsByTag("OpenAI");
```

Comment:
I would suggest to use a tag here that relates to the binding name and not introduce an additional term here:

```suggestion
        Collection<Item> openaiItems = itemRegistry.getItemsByTag("ChatGPT");
```

Reply:
Ok, changed.